src/main/java/nl/tudelft/lifetiles/graph/controller/GraphController.java

```diff
@@ -213,45 +253,88 @@ private void repaint() {
             }
             view = new TileView(this);
 
-            root = new Group();
-
             wrapper.hvalueProperty().addListener(
                     (observable, oldValue, newValue) -> {
-                        repaintPosition(root, newValue.doubleValue());
+                        repaintPosition(newValue.doubleValue());
                     });
 
-            Rectangle clip = new Rectangle(getMaxUnifiedEnd(graph)
-                    * VertexView.HORIZONTALSCALE, 0);
-            root.getChildren().add(clip);
-
-            repaintPosition(root, wrapper.hvalueProperty().doubleValue());
+            repaintPosition(wrapper.hvalueProperty().doubleValue());
         }
     }
 
     /**
      * Repaints the view indicated by the bucket in the given position.
      *
-     * @param root
-     *            Root group used to store the visualized graph in.
      * @param position
      *            Position in the scrollPane.
      */
-    private void repaintPosition(final Group root, final double position) {
-        int nextPosition = getBucketPosition(position);
-        if (currentPosition != nextPosition || forceRepaintPosition) {
+    private void repaintPosition(final double position) {
+        double graphWidth = getMaxUnifiedEnd(graph)
+                * VertexView.HORIZONTALSCALE * scale;
+
+        double screenWidth = wrapper.getViewportBounds().getWidth();
+        double diffWidth = graphWidth - screenWidth;
+        double scaledScreenWidth = screenWidth * 2 * scale;
+
+        double newPosition = (position * (screenWidth / 2 + diffWidth))
```

Comment:
`2` is an `int`, this could cause problems, it might be better to use `2.0` instead.